MvvmCross/ViewModels/IMvxAppStart.cs

```diff
@@ -2,10 +2,14 @@
 // The .NET Foundation licenses this file to you under the MS-PL license.
 // See the LICENSE file in the project root for more information.
 
+using System.Threading.Tasks;
+
 namespace MvvmCross.ViewModels
 {
     public interface IMvxAppStart
     {
         void Start(object hint = null);
+
+        Task WaitForStart();
```

Comment:
I would rather make `void Start(object hint = null);` a `Task`. That might be difficult to make async all the way, so maybe another idea is to create a `IMvxFormsAppStart` for now and make it a Task there or add WaitForStart there. Also we could cast it to make it available.